### PR TITLE
IGNITE-20045 Use classes from Catalog domain in assignments recalculation code

### PR DESCRIPTION
**Pull request by rpuch**
https://issues.apache.org/jira/browse/IGNITE-20045

### CODE REVIEW

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/rebalance/DistributionZoneRebalanceEngineTest.java [hunk 1]
```diff
@@ -443,6 +451,23 @@ void replicasTriggersAssignmentsChangingOnDefaultZone(
         }
     }
 
+    private static void completeTablesConfigs(TablesConfiguration tablesConfiguration) {
```
Comment (sashapolo) [inline modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/rebalance/DistributionZoneRebalanceEngineTest.java:454]: Why is this method needed?

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/ZoneCatalogDescriptorUtils.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ * Utils used to convert configuration representations of distribution zones to catalog representations.
+ */
+public class ZoneCatalogDescriptorUtils {
+    // TODO: IGNITE-19719 Fix it
+    /**
+     * Converts a distribution zone configuration to a Distribution zone descriptor.
+     *
+     * @param config Distribution zone configuration.
+     */
+    @Deprecated(forRemoval = true)
```
Comment (sashapolo) [inline modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/ZoneCatalogDescriptorUtils.java:36]: Looks like the whole class should be deprecated and removed in the future?

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/ZoneCatalogDescriptorUtils.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                toDataStorageDescriptor(config.dataStorage())
+        );
+    }
+
+    @Deprecated(forRemoval = true)
+    private static CatalogDataStorageDescriptor toDataStorageDescriptor(DataStorageView config) {
+        String dataRegion;
+
+        try {
+            Method dataRegionMethod = config.getClass().getMethod("dataRegion");
```
Comment (sashapolo) [inline modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/ZoneCatalogDescriptorUtils.java:56]: oh god...

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/rebalance/DistributionZoneRebalanceEngineTest.java (hunk 1 above)
Comment (rpuch) [reply modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/rebalance/DistributionZoneRebalanceEngineTest.java:454]: The code that processes rebalance converts `TableView` to `CatalogTableDescriptor` and fails if there is no primary key or columns, so it is a temporary ceremony that will go away when we switch to the Catalog completely

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/ZoneCatalogDescriptorUtils.java (hunk 3 above)
Comment (rpuch) [reply modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/ZoneCatalogDescriptorUtils.java:56]: I agree!

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/ZoneCatalogDescriptorUtils.java (hunk 2 above)
Comment (rpuch) [reply modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/ZoneCatalogDescriptorUtils.java:37]: Deprecated the class too

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/rebalance/DistributionZoneRebalanceEngineTest.java (hunk 1 above)
Comment (sashapolo) [reply modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/rebalance/DistributionZoneRebalanceEngineTest.java:454]: can it be provided as part of the initial configuration inside `@InjectConfiguration`?

**Review by sashapolo — APPROVED** [review]

**Pull request merged**